sdk/python/kfp/compiler/compiler.py

```diff
@@ -638,7 +638,8 @@ def _create_dag_templates(self, pipeline, op_transformers=None, op_to_templates_
       templates.append(template)
 
     for op in pipeline.ops.values():
-      templates.extend(op_to_templates_handler(op))
+      op_to_template_handler = op.template_handler or op_to_templates_handler
```

Comment:
Would it be possible to add a short comment on what `op.template_handler` and `op_to_templates_handler` are? We have 3 different variables here which look very similar to each other, so a comment would definitely help a reader understand what's doing on:)